src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Double.cs

```diff
@@ -88,9 +90,55 @@ private void WriteNumberValueIndented(double value)
                 BytesPending += indent;
             }
 
-            bool result = Utf8Formatter.TryFormat(value, output.Slice(BytesPending), out int bytesWritten);
+            bool result = TryFormatDouble(value, output.Slice(BytesPending), out int bytesWritten);
             Debug.Assert(result);
             BytesPending += bytesWritten;
         }
+
+        private static bool TryFormatDouble(double value, Span<byte> destination, out int bytesWritten)
+        {
+            // Frameworks that are not .NET Core 3.0 or higher do not produce roundtrippable strings by
+            // default. Further, the Utf8Formatter on older frameworks does not support taking a precision
+            // specifier for 'G' nor does it represent other formats such as 'R'. As such, we duplicate
+            // the .NET Core 3.0 logic of forwarding to the UTF16 formatter and transcoding it back to UTF8,
+            // with some additional changes to remove dependencies on Span APIs which don't exist downlevel.
```

Comment:
Which APIs are those? It is unfortunate that we end up calling `value.ToString()` and allocating a string, for downlevel, which is then transcoded to UTF-8 (especially when `Utf8Formatter` exists down-level). If there is no other feasible solution, can you link to the servicing fix issue in a comment that will let us avoid the extra allocation once that's in?

Reply:
Full framework and netstandard dont expose any of the Span APIs that arent extension methods, so basically everything on string and UTF8Encoding

Reply:
Note this is also how Utf8Formatter shipped in .NET Core 2.1.  It wasn't until https://github.com/dotnet/coreclr/pull/22011 that it changed, so it can't have been that problematic.

Reply:
I think it will be fine for .NET 2.1/2.2 and Full Framework. Without the change, then the number of uniquely serializable doubles is effectively a third of what it should be.